core/src/main/scala/tech/beshu/ror/accesscontrol/factory/decoders/AuditingSettingsDecoder.scala

```diff
@@ -37,18 +39,83 @@ object AuditingSettingsDecoder extends Logging {
   lazy val instance: Decoder[Option[AuditingTool.Settings]] =
     Decoder.instance { c =>
       whenEnabled(c) {
-        for {
-          auditIndexTemplate <- decodeOptionalSetting[RorAuditIndexTemplate](c)("index_template", fallbackKey = "audit_index_template")
-          customAuditSerializer <- decodeOptionalSetting[AuditLogSerializer](c)("serializer", fallbackKey = "audit_serializer")
-          remoteAuditCluster <- decodeOptionalSetting[AuditCluster.RemoteAuditCluster](c)("cluster", fallbackKey = "audit_cluster")
-        } yield AuditingTool.Settings(
-          auditIndexTemplate.getOrElse(RorAuditIndexTemplate.default),
-          customAuditSerializer.getOrElse(new DefaultAuditLogSerializer),
-          remoteAuditCluster.getOrElse(AuditCluster.LocalAuditCluster)
-        )
+        Decoder[AuditingTool.Settings].apply(c)
       }
     }
 
+  private implicit val auditSinkConfigsDecoder: Decoder[AuditingTool.Settings] = Decoder.instance { c =>
+    for {
+      auditIndexTemplate <- decodeOptionalSetting[RorAuditIndexTemplate](c)("index_template", fallbackKey = "audit_index_template")
+      customAuditSerializer <- decodeOptionalSetting[AuditLogSerializer](c)("serializer", fallbackKey = "audit_serializer")
+      remoteAuditCluster <- decodeOptionalSetting[AuditCluster.RemoteAuditCluster](c)("cluster", fallbackKey = "audit_cluster")
+      maybeOutputs <- c.downField("audit").downField("outputs").as[Option[List[AuditSinkConfig]]]
+      sinkConfigs <- maybeOutputs match {
+        case Some(value) =>
+          for {
+            _ <- Either.cond(
+              List(auditIndexTemplate, customAuditSerializer, remoteAuditCluster).forall(_.isEmpty),
+              (),
+              DecodingFailure(
+                message = "Audit config contains 'outputs' section and deprecated audit properties." +
```

Comment:
I am not sure if we should extend the `audit` section or consider the `v2` variant. The `audit` section contains many fallback keys and becomes harder to maintain. I will supply the rest of the unit tests when we make the decision

Reply:
ok, we can provide v2 variant, but under a few assumptions:
* backward compatibility for in-index audit
* the new type of sink can be supported in v2 only (if users want to use it, they should adapt the v2)
* from the perspective of code we don't need to persist the information that we handle two versions of configuration 
* we should fix all audit examples in our docs to the format